File: 1677-matrix-diagonal-sum/matrix-diagonal-sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int sum=0;
        int n=mat.size();
        for(int i=0; i<n; i++){
            sum+= mat[i][i]; // 00 11  22

            sum+= mat[i][n-1-i]; // 30 ,22 , 
        }

        if(n%2!= 0){
            sum -= mat[n/2][n/2];
        }
        return sum;
    }
};

File: 1677-matrix-diagonal-sum/matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int m=mat.length;
        int n=mat[0].length;
        int sum=0;

        for(int i=0;i<m; i++){
           sum+=mat[i][i];
           sum+=mat[i][n-1-i];
        }
        if(n%2!=0){
        sum-= mat[n/2][n/2];
        }
        return sum;
    }
}